internal/xcrd/crd.go

```diff
@@ -144,53 +111,71 @@ func ForCompositeResourceClaim(xrd *v1.CompositeResourceDefinition) (*extv1.Cust
 	crd.Spec.Names.Categories = append(crd.Spec.Names.Categories, CategoryClaim)
 
 	for i, vr := range xrd.Spec.Versions {
-		crd.Spec.Versions[i] = extv1.CustomResourceDefinitionVersion{
-			Name:                     vr.Name,
-			Served:                   vr.Served,
-			Storage:                  vr.Referenceable,
-			Deprecated:               pointer.BoolDeref(vr.Deprecated, false),
-			DeprecationWarning:       vr.DeprecationWarning,
-			AdditionalPrinterColumns: append(vr.AdditionalPrinterColumns, CompositeResourceClaimPrinterColumns()...),
-			Schema: &extv1.CustomResourceValidation{
-				OpenAPIV3Schema: BaseProps(),
-			},
-			Subresources: &extv1.CustomResourceSubresources{
-				Status: &extv1.CustomResourceSubresourceStatus{},
-			},
-		}
-
-		p, required, err := getProps("spec", vr.Schema)
+		crdv, err := genCrdVersion(vr)
 		if err != nil {
-			return nil, errors.Wrapf(err, errFmtGetProps, "spec")
-		}
-		specProps := crd.Spec.Versions[i].Schema.OpenAPIV3Schema.Properties["spec"]
-		specProps.Required = append(specProps.Required, required...)
-		for k, v := range p {
-			specProps.Properties[k] = v
+			return nil, err
```

Comment:
I'm still trying to figure out why we wrap errors. If the only reason is to provide useful context about the error, I don't think it's necessary here, since all the errors in `genCrdVersion` are already wrapped with context, either in that function, or in `parseSchema`. Is there a reason to add an additional layer of wrapping? 

And what would I wrap it with? Something like "error generating custom resource definition for xrd spec version index `i`" (defined in the constants like the other messages, of course)

Reply:
@mbbush not sure if you're aware but we recently updated the contribution guides around errors: https://github.com/crossplane/crossplane/tree/master/contributing#wrap-errors and https://github.com/crossplane/crossplane/blob/master/contributing/guide-observability.md#error-reporting-as-events

Reply:
Thanks @jeanduplessis. I'd seen the doc about wrapping errors before, but not the event one. 

Is the idea that we're building a sort of stacktrace by adding an additional layer of wrapping every time there's a function return? 

I'm happy to wrap each error an additional time, I'd just really like to understand why so that I can write a helpful message.

Reply:
Hi @mbbush, `genCrdVersion` is called in two different functions and wrapping the error will help us to distinguish. Also, it is a good practice to wrap errors each time we return for future-proofing as well.

Reply:
Thanks, that's helpful context, and helps me think of a good message to put in the wrapper, too. Will do.